src/main/java/dev/dokan/dokan_java/structure/DokanControl.java

```diff
@@ -79,19 +83,28 @@ public DokanControl(Pointer p, long offset) {
      * @param length the number of elements in the array. Also acquired with the native method call.
      * @return a list of DokanControl structures
      */
-    public static List<DokanControl> getDokanControlList(Pointer start, long length) {
+    public static List<DokanControl> getDokanControlList(Pointer start, @Unsigned int length) { //TODO Relocate
         List<DokanControl> list = new ArrayList<>();
-        if (length == 0) {
-            return list;
-        } else if (length < 0) {
-            //TODO length is actually an unsigned long! -> java always treats them as signed
-            return list;
-        } else {
-            list.add(new DokanControl(start));
-            for (int i = 1; i < length; i++) {
-                list.add(new DokanControl(start, i * list.get(0).size()));
+        /*
+         * Let's do the math:
+         * A list that uses an unsigned int (32 bit) as index could save up to 2^32 objects.
+         * Even if it only saved one byte in each entry (not accounting for the actual overhead of the entries themselves),
+         * that would be 2^32 bytes = 4 GB for that list alone!
+         * If the list of active Dokan MountPoints exceeds 2^31 (signed int) entries, we probably have other problems.
+         *
+         * But let's assume that this could still happen:
+         * In this case we want the application to crash so that someone else can fix that nonsense.
+         * "assert" seems like a good choice for this case (for once).
+         */
```

Comment:
As discussed: Less sassy, and move it the javadocs part for documentation purpose.

Reply:
Resolved by commit 279b5b890b4b148a86fe25ad1d9b43f9fd7d8507